signer/threshold_signer_soft.go

```diff
@@ -0,0 +1,354 @@
+package signer
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	tmcryptoed25519 "github.com/tendermint/tendermint/crypto/ed25519"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	"gitlab.com/unit410/edwards25519"
+	tsed25519 "gitlab.com/unit410/threshold-ed25519/pkg"
+)
+
+// ThresholdSignerSoft implements the interface and signs the message for each local signer.
+// ThresholdSignerSoft is the implementation of a soft sign signer at the local level.
+type ThresholdSignerSoft struct {
+	PubKeyBytes []byte
+	Key         CosignerKey
+	// Total signers
+	Total     uint8
+	Threshold uint8
+	// Height, Round, Step, Timestamp --> metadata
+	HrsMeta map[HRSTKey]HrsMetadata
+}
+
+// NewThresholdSignerSoft constructs a ThresholdSigner
+// that signs using the local key share file.
+func NewThresholdSignerSoft(
+	key CosignerKey,
+	threshold, total uint8,
+) ThresholdSigner {
+	softSigner := &ThresholdSignerSoft{
+		Key:       key,
+		HrsMeta:   make(map[HRSTKey]HrsMetadata),
+		Total:     total,
+		Threshold: threshold,
+	}
+
+	// cache the public key bytes for signing operations
+	switch ed25519Key := softSigner.Key.PubKey.(type) {
```

Comment:
lets replace this switch case with something like
```golang
ed25519Key := softSigner.Key.PubKey.(tmcryptoed25519.PubKey)
softSigner.PubKeyBytes = make([]byte, len(ed25519Key))
copy(softSigner.PubKeyBytes, ed25519Key[:])
```

As @mark-rushakoff pointed out in the larger PR for a similar issue, the default panic for a failure to cast to `tmcryptoed25519.PubKey` will be better since it will include details about the incorrect type.

Reply:
Should be fixed now.